tests/unit/amo/components/TestInstallButtonWrapper.js

```diff
@@ -820,19 +770,990 @@ describe(__filename, () => {
       // individual tests above test separate pieces of logic.
       it('returns the expected URL for an add-on', () => {
         const addonId = 123;
-        const addon = createInternalAddonWithLang({
+        const internalAddon = createInternalAddonWithLang({
           ...fakeAddon,
           id: addonId,
+          guid,
         });
 
         const expectedLink = [
           `${DOWNLOAD_FIREFOX_BASE_URL}?s=direct`,
           `utm_campaign=${DOWNLOAD_FIREFOX_UTM_CAMPAIGN}-${addonId}`,
-          `utm_content=rta%3A${encode(addon.guid)}`,
+          `utm_content=rta%3A${encode(guid)}`,
           `utm_medium=referral&utm_source=addons.mozilla.org`,
         ].join('&');
 
-        expect(getDownloadLink({ addon })).toEqual(expectedLink);
+        expect(getDownloadLink({ addon: internalAddon })).toEqual(expectedLink);
+      });
+    });
+  });
+
+  describe('Tests for withInstallHelpers', () => {
+    it('calls getAddon() when the component is rendered', () => {
+      renderWithCurrentVersion();
+
+      expect(_addonManager.getAddon).toHaveBeenCalledWith(fakeAddon.guid);
+    });
+
+    it('does not call getAddon() if we do not have an addonManager', () => {
+      renderWithCurrentVersion({
+        addonManagerOverrides: { hasAddonManager: false },
+      });
+
+      expect(_addonManager.getAddon).not.toHaveBeenCalled();
+    });
+
+    it('does not call getAddon() if we do not have an addon', () => {
+      render({ addon: null });
+
+      expect(_addonManager.getAddon).not.toHaveBeenCalled();
+
+      // We didn't render a proper install button wrapper.
+      expect(
+        screen.queryByClassName('InstallButtonWrapper'),
+      ).not.toBeInTheDocument();
+    });
+
+    describe('setCurrentStatus', () => {
+      it('sets the status to ENABLED when an extension is enabled', async () => {
+        const installURL = fakeAddon.current_version.file.url;
```

Comment:
This should now use `addon` instead of `fakeAddon`. I'm going to stop adding these comments. For the most part, anywhere in the code that is currently using `fakeAddon` should probably be changed to use `addon`.